### PR TITLE
Improve watch()

### PR DESCRIPTION
**Pull request by nickpape-msft**
* We were experiencing an issue where, if there was a build error, "gulp serve" was ending instead of continuing to watch files for changes.
* This was because we had defined the build tasks twice, once in the watch() block, but again in a serial() block enveloping the two.
* After removing the duplicate build steps, I noticed the issue was that watch() was not kicked off unless there was a file change. Therefore, watch() was changed so that it would always initially invoke itself. 
* While doing that I noticed some issues with the promises (e.g. watch was reporting to gcb that it was completing before it actually was), so I fixed that issue by removing the `return Promise.resolve()` and replacing it with a Promise block that never returns.

### CODE REVIEW

**Comment by nickpape-msft** [comment]
Should fix https://github.com/SharePoint/sp-dev-docs/issues/320

File: core-build/gulp-core-build/src/index.ts [hunk 1]
```diff
@@ -188,57 +189,71 @@ export function watch(watchMatch: string | string[], task: IExecutable): IExecut
...
-                  title: 'Build succeeded',
-                  message: packageJSON.name,
-                  icon: buildConfig.buildSuccessIconPath
-                });
-              }
-              _finalizeWatch();
-            })
-            .catch((error) => {
-              if (buildConfig.showToast) {
+      return new Promise<void>((resolve: () => void, reject: () => void) => {
```
Comment (iclanton) [inline core-build/gulp-core-build/src/index.ts]: >(resolve: () => void, reject: () => void) [](start = 31, length = 41)

You don't need these parameters.

File: core-build/web-library-build/src/index.ts [hunk 2]
```diff
@@ -39,7 +39,7 @@ const sourceMatch: string[] = [
   '!src/**/*.scss.ts'
 ];
 
-const PRODUCTION = process.argv.indexOf('--production') !== -1 || process.argv.indexOf('--ship') !== -1;
+const PRODUCTION = !!getConfig().args['--production'] || !!getConfig().args['--ship'];
```
Comment (iclanton) [inline core-build/web-library-build/src/index.ts]: >-- [](start = 39, length = 2)

You don't need the dashes here.

**Comment by iclanton** [comment]
:shipit:

**Pull request merged**